test/namedtuple/productnamedtuple.jl

```diff
@@ -0,0 +1,195 @@
+using Distributions
+using Distributions: ProductNamedTupleDistribution, ProductNamedTupleSampler
+using LinearAlgebra
+using Random
+using Test
+
+@testset "ProductNamedTupleDistribution" begin
+    @testset "Constructor" begin
+        nt = (x=Normal(1.0, 2.0), y=Normal(3.0, 4.0))
+        d = @inferred ProductNamedTupleDistribution(nt)
+        @test d isa ProductNamedTupleDistribution
+        @test d.dists === nt
+        @test Distributions.variate_form(typeof(d)) === NamedTupleVariate{(:x, :y)}
+        @test Distributions.value_support(typeof(d)) === Continuous
+
+        nt = (
+            x=Normal(),
+            y=Dirichlet(10, 1.0),
+            z=DiscreteUniform(1, 10),
+            w=LKJCholesky(3, 2.0),
+        )
+        d = @inferred ProductNamedTupleDistribution(nt)
+        @test d isa ProductNamedTupleDistribution
+        @test d.dists === nt
+        @test Distributions.variate_form(typeof(d)) === NamedTupleVariate{(:x, :y, :z, :w)}
+        @test Distributions.value_support(typeof(d)) === Continuous
+    end
+
+    @testset "product_distribution" begin
+        nt = (x=Normal(1.0, 2.0), y=Normal(3.0, 4.0))
+        d = @inferred product_distribution(nt)
+        @test d === ProductNamedTupleDistribution(nt)
+
+        nt = (
+            x=Normal(),
+            y=Dirichlet(10, 1.0),
+            z=DiscreteUniform(1, 10),
+            w=LKJCholesky(3, 2.0),
+        )
+        d = @inferred product_distribution(nt)
+        @test d === ProductNamedTupleDistribution(nt)
+    end
+
+    @testset "show" begin
+        d = ProductNamedTupleDistribution((x=Gamma(1.0, 2.0), y=Normal()))
+        @test sprint(show, d) == """
```

Comment:
FYI an alternative would be
```suggestion
        @test repr(d) == """
```